src/celeritas/ext/VecgeomTrackView.hh

```diff
@@ -390,7 +390,7 @@ CELER_FUNCTION void VecgeomTrackView::cross_boundary()
 CELER_FUNCTION void VecgeomTrackView::move_internal(real_type dist)
 {
     CELER_EXPECT(this->has_next_step());
-    CELER_EXPECT(dist > 0 && dist <= next_step_);
+    CELER_EXPECT(dist > 0 && dist <= next_step_ + real_type(1e-16));
```

Comment:
If `next-step_` is greater than or equal to 1, or if we're using single precision and the step is greater than 1e-8,  then this will have no effect... since the case you saw this was with a step of `O(1e-8)` (https://github.com/celeritas-project/celeritas/pull/484#issuecomment-1218922244) maybe this should be
```suggestion
    CELER_EXPECT(dist > 0 && dist <= next_step_ * real_type(1 + 1e-8));
```

Reply:
This brings me to the float vs. double issue.  These 1e-8 or 1e-16 factors make sense for double only, but the real_type can be an alias for float, in which case the 1e-8 does not make any sense.  
VecGeom uses different tolerance values depending on the precision.  Shouldn't celeritas do some similar tuning to these ad-hoc values?

Reply:
I was thinking about that too: I don't know whether the `1e-8` needs to be `1e-4` with if it's floats, or what, but there might need to be a floating-point traits like we have with `SoftEquivTraits`. I think for bumps, though, multiplicative is more likely to be correct than additive.

Reply:
Looking at the SoftEqual* code, I don't see (yet at least) how to use it for the `dist <= next_step_` case, when `dist` is not really close to `next_step`.

Reply:
I have added a `soft_equal(dist, next_step_)` to the condition.